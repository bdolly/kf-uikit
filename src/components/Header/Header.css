.Header {
    @apply
      w-full
      bg-white;
}

.Header--alert {
    @apply
      px-2
      py-0
      bg-primary
      border-b-2
      border-primary-dark
      text-white
      font-light;
}

@screen md {
  .Header--alert {
    @apply text-center;
  }
}

.Header--navigation {
    @apply
      flex
      h-24
      p-4
      border-b-2;
}

.Navigation--buttons {
    @apply
      flex
      items-center
      justify-end
      w-full;
}


.Header--navigation .Button {
    @apply
        mx-2;
}
