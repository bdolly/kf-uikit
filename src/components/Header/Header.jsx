import React from 'react';
import propTypes from 'prop-types';
import className from 'classnames';
import Logo from '../Logo/Logo';
import './Header.css';

/**
 * The common Kids First navigation.
 *
 * Must pass an array of buttons for the navigation bar.
 * Providing alertMessage will include an alert at the top of the header.
 *
 * The header should be included at the top of every Kids First application.
 * It contains common functionality for all of Kids-First that should remain
 * consistent, such as the user drop down and application navigation.
 */
const Header = ({ buttons, alertMessage, children }) => {

  const navClass = className(
    'Header--navigation',
  );

  return (
    <header className='Header'>
      {alertMessage && (
        <div className='Header--alert'>
          {alertMessage}
        </div>
      )}
      <nav className={navClass}>
        <Logo />
        <div className='Navigation--buttons'>
          {buttons ? buttons.map(button => button) : null}
        </div>
        {children}
      </nav>
    </header>
  );
}

Header.propTypes = {
  /** Buttons to display on the navbar */
  buttons: propTypes.node,
  /** Alert message to be displayed on top of the page */
  alertMessage: propTypes.string,
  /** Child elements to display on the navbar */
  children: propTypes.node,
};

Header.defaultProps = {
  buttons: null,
  alertMessage: null,
  children: null,
};

export default Header;
