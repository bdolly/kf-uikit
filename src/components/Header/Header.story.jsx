/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text } from '@storybook/addon-knobs';
import Button from '../Button/Button';
import Header from './Header';

const stories = storiesOf('Header', module);

stories.addDecorator(withKnobs);

stories.add('Header', () => (
  <Header
    alertMessage={text('Alert message', 'New data available!')}
    buttons={[
      <Button>Dashboard</Button>,
      <Button outline>File Repository</Button>]}
  />
));
