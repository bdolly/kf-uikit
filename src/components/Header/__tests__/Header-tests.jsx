import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../Header';

it(`navigation renders correctly`, () => {
    const tree = renderer.create(
      <Header />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it(`header with alert renders correctly`, () => {
    const tree = renderer.create(
      <Header
        alertMessage='Testing'
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
