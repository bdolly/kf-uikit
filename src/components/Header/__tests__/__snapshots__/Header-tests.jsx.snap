// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header with alert renders correctly 1`] = `
<header
  className="Header"
>
  <div
    className="Header--alert"
  >
    Testing
  </div>
  <nav
    className="Header--navigation"
  >
    <div
      className="Logo"
    >
      <img
        alt="Kids First Data Resource Center logo"
        className="Logo--img"
        src="test-file-stub"
      />
      <h1
        className="Logo--text text-screen-reader"
      >
        Kids First Data Resource Center logo
      </h1>
    </div>
    <div
      className="Navigation--buttons"
    />
  </nav>
</header>
`;

exports[`navigation renders correctly 1`] = `
<header
  className="Header"
>
  <nav
    className="Header--navigation"
  >
    <div
      className="Logo"
    >
      <img
        alt="Kids First Data Resource Center logo"
        className="Logo--img"
        src="test-file-stub"
      />
      <h1
        className="Logo--text text-screen-reader"
      >
        Kids First Data Resource Center logo
      </h1>
    </div>
    <div
      className="Navigation--buttons"
    />
  </nav>
</header>
`;
